lua/clipboard-image/paste.lua

```diff
@@ -15,20 +11,20 @@ M.paste_img = function(opts)
     return false
   end
 
+  local conf_toload = conf_utils.get_usable_config()
+  conf_toload = conf_utils.merge_config(conf_toload, opts)
+  local conf = conf_utils.load_config(conf_toload)
+
   local content = utils.get_clip_content(cmd_check)
-  if utils.is_clipboard_img(content) ~= true then
+
+  if utils.is_clipboard_img(content, conf.img_format) ~= true then
     vim.notify("There is no image data in clipboard", vim.log.levels.ERROR)
   else
-    local conf_toload = conf_utils.get_usable_config()
-    conf_toload = conf_utils.merge_config(conf_toload, opts)
-
-    local conf = conf_utils.load_config(conf_toload)
-    local path = utils.get_img_path(conf.img_dir, conf.img_name)
-    local path_txt = utils.get_img_path(conf.img_dir_txt, conf.img_name, "txt")
+    local path = utils.get_img_path(conf.img_dir, conf.img_name, conf.img_ext)
+    local path_txt = utils.get_img_path(conf.img_dir_txt, conf.img_name, conf.img_ext, "txt")
```

Comment:
I think it's probably better to let the method `get_img_path` generate the whole path to avoid splitting the path generating logic and such that it can be used elsewhere if needed.